mono/utils/mono-math-c.c

```diff
@@ -0,0 +1,22 @@
+// This file bridges C++ to C runtime math functions
+// without any chance of a C++ library dependency.
+// In time it can probably be removed.
+#include "config.h"
+#include "glib.h"
+#define MONO_MATH_DECLARE_ALL 1
+#include "mono-math.h"
+
+#if defined (__cplusplus) && !defined (_MSC_VER)
```

Comment:
fixme: Project file work to compile as C also for Visual C++? But it doesn't really matter. This work is mostly for Android and WebAssembly. Mac/Windows/Linux were all building ok previously. iOS uncertain.
What put me over the edge to not just keep using "using" is the symbols imported on Linux in a toy example -- mangled isinf or such did appear. This should avoid that.
Eventually maybe the inline functions in mono-math.h can just call std::foo but this approach is definitely a little safer, for the cost of roughly an instruction per call to these functions.